diff --git a/CMakeLists.txt b/CMakeLists.txt
index c9861d4..7fef403 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,14 +34,6 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 # Turn on generating compile_commands.json for clang-tidy and iwyu.
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-if (NOT CMAKE_SYSTEM_PROCESSOR)
-	message(FATAL_ERROR "No target CPU architecture set")
-endif()
-
-if (NOT CMAKE_SYSTEM_NAME)
-	message(FATAL_ERROR "No target OS set")
-endif()
-
 # Set CMAKE_BUILD_TYPE if unset.
 include(BuildType)
 message (STATUS "CMAKE_BUILD_TYPE = ${CMAKE_BUILD_TYPE}")
